sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableSpec.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import org.apache.spark.SparkThrowable
+import org.apache.spark.sql.catalyst.expressions.{Expression, Literal}
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.catalyst.trees.TreePattern.COMMAND
+import org.apache.spark.sql.errors.QueryCompilationErrors
+import org.apache.spark.sql.types.{ArrayType, MapType, StructType}
+
+/**
+ * This object is responsible for processing unresolved table specifications in commands with
+ * OPTIONS lists. The parser produces such lists as maps from strings to unresolved expressions.
+ * After otherwise resolving such expressions in the analyzer, here we convert them to resolved
+ * table specifications wherein these OPTIONS list values are represented as strings instead, for
+ * convenience.
+ */
+object ResolveTableSpec extends Rule[LogicalPlan] {
+  override def apply(plan: LogicalPlan): LogicalPlan = {
+    plan.resolveOperatorsWithPruning(_.containsAnyPattern(COMMAND), ruleId) {
+      case t: CreateTable =>
+        resolveTableSpec(t, s => t.copy(tableSpec = s))
+      case t: CreateTableAsSelect =>
+        resolveTableSpec(t, s => t.copy(tableSpec = s))
+      case t: ReplaceTable =>
+        resolveTableSpec(t, s => t.copy(tableSpec = s))
+      case t: ReplaceTableAsSelect =>
+        resolveTableSpec(t, s => t.copy(tableSpec = s))
+    }
+  }
+
+  /** Helper method to resolve the table specification within a logical plan. */
+  private def resolveTableSpec(
+      t: HasTableSpec, withNewSpec: TableSpec => LogicalPlan): LogicalPlan = t.tableSpec match {
+    case u: UnresolvedTableSpec if t.unresolvedOptionsList.allOptionsResolved =>
+      val newOptions: Seq[(String, String)] = t.unresolvedOptionsList.options.map {
+        case (key: String, null) =>
+          (key, null)
+        case (key: String, value: Expression) =>
+          val newValue: String = try {
+            val dt = value.dataType
+            value match {
+              case Literal(null, _) =>
+                "null"
```

Comment:
should it be `null` or `"null"`?

Reply:
It should be "null", this part computes the string value for the table option.

Reply:
But string can be null too. If it is `"null"`, then the literal should be `Literal("null", StringType)`